code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm

```diff
@@ -106,13 +106,31 @@
 	metabolization_rate = 0.05
 	taste_description = "salt"
 
+/datum/reagent/blackpowder/on_mob_life(mob/living/M)
+	M.losebreath += 10
+	if(holder.has_reagent("blackpowder"))
```

Comment:
Why are you checking if holder has blackpowder in on_mob_life? Is that not implied?